pkgs/games/ja2-stracciatella/default.nix

```diff
@@ -1,45 +1,64 @@
-{ stdenv, fetchFromGitHub, cmake, SDL2, boost, fltk, rustPlatform }:
+{ stdenv, lib, fetchurl, fetchFromGitHub, cmake, python, rustPlatform, SDL2, fltk, rapidjson, Carbon, Cocoa }:
 let
-  version = "0.16.1";
+  version = "0.17.0";
   src = fetchFromGitHub {
     owner = "ja2-stracciatella";
     repo = "ja2-stracciatella";
     rev = "v${version}";
-    sha256 = "1pyn23syg70kiyfbs3pdlq0ixd2bxhncbamnic43rym3dmd52m29";
-  };
-  lockfile = ./Cargo.lock;
-  libstracciatellaSrc = stdenv.mkDerivation {
-    name = "libstracciatella-${version}-src";
-    src = "${src}/rust";
-    installPhase = ''
-      mkdir -p $out
-      cp -R ./* $out/
-      cp ${lockfile} $out/Cargo.lock
-    '';
+    sha256 = "0m6rvgkba29jy3yq5hs1sn26mwrjl6mamqnv4plrid5fqaivhn6j";
   };
   libstracciatella = rustPlatform.buildRustPackage {
     pname = "libstracciatella";
     inherit version;
-    src = libstracciatellaSrc;
-    cargoSha256 = "15djs4xaz4y1hpfyfqxdgdasxr0b5idy9i5a7c8cmh0jkxjv8bqc";
-    doCheck = false;
+    src = "${src}/rust";
+    cargoSha256 = "0blb971cv9k6c460mwq3zq8vih687bdnb39b9gph1hr90pxjviba";
+
+    preBuild = ''
+      mkdir -p $out/include/stracciatella
+      export HEADER_LOCATION=$out/include/stracciatella/stracciatella.h
+    '';
+  };
+  stringTheoryUrl = https://github.com/zrax/string_theory/archive/3.1.tar.gz;
+  stringTheory = fetchurl {
+    url = stringTheoryUrl;
+    sha256 = "1flq26kkvx2m1yd38ldcq2k046yqw07jahms8a6614m924bmbv41";
+  };
+  gtestUrl = https://github.com/google/googletest/archive/release-1.10.0.tar.gz;
+  gtest = fetchurl {
+    url = gtestUrl;
+    sha256 = "1jvdf56gcvhyi9fj70gcigxva2sad6dfmnj3grxfql8mk9x1bjcx";
```

Comment:
Can we use https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/gtest/default.nix ?

Reply:
Definitely. Small question: Do you know why it is now showing in the [nix package search](https://search.nixos.org/packages?channel=20.09&from=0&size=50&sort=relevance&query=gtest)?

Reply:
I think gtest is only available in unstable. That shouldn't be an issue for this PR though.